From: Quinn Tran <qutran@marvell.com>
Date: Wed, 26 Feb 2020 14:40:17 -0800
Subject: scsi: qla2xxx: Fix NPIV instantiation after FW dump
Patch-mainline: v5.7-rc1
Git-commit: ab391abdc191c2a2874098df86d1fe4b9762ef60
References: bsc#1157424

NPIV re-enable code was blocked after FW had been initialized.  The
blocking check was too broad. Reduce the check to make sure if FW is
initialized or not.

Link: https://lore.kernel.org/r/20200226224022.24518-14-hmadhani@marvell.com
Signed-off-by: Himanshu Madhani <hmadhani@marvell.com>
Signed-off-by: Quinn Tran <qutran@marvell.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Daniel Wagner <dwagner@suse.de>
---
 drivers/scsi/qla2xxx/qla_mbx.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/scsi/qla2xxx/qla_mbx.c
+++ b/drivers/scsi/qla2xxx/qla_mbx.c
@@ -1387,7 +1387,7 @@ qla2x00_issue_iocb_timeout(scsi_qla_host
 	mbx_cmd_t	mc;
 	mbx_cmd_t	*mcp = &mc;
 
-	if (qla2x00_chip_is_down(vha))
+	if (!vha->hw->flags.fw_started)
 		return QLA_INVALID_COMMAND;
 
 	ql_dbg(ql_dbg_mbx + ql_dbg_verbose, vha, 0x1038,
